lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -879,6 +880,13 @@ sub investigate {
     my ($self) = @_;
     return $self->reply->not_found unless my $job = $self->get_current_job;
     my $investigation = $job->investigate;
+    my $last_good_id = ${$investigation}{'last_good'};
+
+    if (looks_like_number($last_good_id)) {
+        delete(${$investigation}{'last_good'});
+        ${$investigation}{'last_good:link'} = {'link' => '/test/'.$last_good_id, 'text' => $last_good_id};
+    }
+
```

Comment:
I would prefer if we can put that directly into the `lib/OpenQA/Schema/Result/Jobs.pm:sub investigate` for two reasons:

* Anyone using the API can benefit from having a directly clickable URL as well
* We can keep the method here trivial without complicated business logic. This is also the reason why currently there is no need for a direct test of this method as we can rely on the test of the low-level method `lib/OpenQA/Schema/Result/Jobs.pm:sub investigate`

Reply:
Ok, this was a doubt I had. I thought that putting this here wouldn't break anything in the API. But is all the tests cover this part of the API I don't see any reason to move this to API

Reply:
Not sure what you mean, with "I don't see any reason to move this to API". I suggested to move it to the result class definitions. This way the functionality is moved both in the REST API as well as the web UI. Maybe you meant "any reason to *not* move"? :)

Reply:
Yes, " any reason to not move"

Reply:
Done